petastorm/etl/dataset_metadata.py

```diff
@@ -98,10 +98,14 @@ def materialize_dataset(spark, dataset_url, schema, row_group_size_mb=None, use_
     # After job completes, add the unischema metadata and check for the metadata summary file
     if pyarrow_filesystem is None:
         resolver = FilesystemResolver(dataset_url, spark.sparkContext._jsc.hadoopConfiguration())
+        fs_factory = resolver.filesystem_factory()
         filesystem = resolver.filesystem()
         dataset_path = resolver.get_dataset_path()
     else:
+        # If we are given a filesystem, then the factory
+        # is that filesystem's constructor
         filesystem = pyarrow_filesystem
+        fs_factory = type(filesystem)
```

Comment:
This is not going to work in general case. The construction of the filesystem object might be taking parameters provided by user. I would expect user to pass a factory and not an instance for this to work properly.

Reply:
Changed.